app.js

```diff
@@ -1,8 +1,17 @@
-const hashToTest = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a'
 const hashString = 'Hello from IPFS Gateway Checker'
+var protocolToTest = 'ipfs'
+var hashToTest = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a'
```

Comment:
Same ↑

Reply:
I think they can't be consts because they can change depending on the URL parameters. Both the protocol and the hashToTest.

Reply:
Can't we change the `hashString` too?
As only this `Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a` should always be "Hello from IPFS Gateway Checker"

Reply:
Do you mean that hashString should be a var and URL parameter as well?

My understanding is that hashString can stay as const, because the only hash content we know for sure is the content of `Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a` as you said. In all the other cases we have no idea what the content will be.

Reply:
Later we test:
```
const matched = text.trim() === hashString.trim()`
const status = matched ? 'All good' : 'Output did not match expected output'
[...]
if(matched) {
     node.innerHTML = '✅ - Online  - ' + returnHtmlLink(gateway)
}
```

Reply:
So having a different hashString than the one equals the gateway is considered as offline

Reply:
I remembered as if we just showed a warning if the content was not matched, but let me check it real quick. You are probably right, and I need to put an extra check there before we verify the content. (Content checking should be skipped for custom hashes.)

Reply:
Yes, you are definitely right :) I'm a Node.js newbie, so is there a way to use the fetch in a way that we limit the amount of data we would like to receive? 
Otherwise it would take a huge amount of time if we wanted to fetch a big file just to realize that it's not the `Hello from ...` message. 
On the other hand we need to fetch something in order to check if the custom hash lookup works, right?

Reply:
What do you think @jamiew and @victorb?

Reply:
After reading about the [fetch method](https://javascript.info/fetch) a little, theoretically we could first try to use the `HEAD` HTTP method (instead of the `GET`) which would return the `Content-Length` header, indicating the size of the content. If we get an OK status, then the node must be online, if the content length were small, we could check the content too.

Unfortunately despite my tested IPFS node responses to both HEAD and GET, neither the HEAD, nor the GET method returns the `Content-Length` header.